addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3837,8 +3838,9 @@ export class OdooEditor extends EventTarget {
             }
             // Find next character.
             let nextCharacter = focusNode.textContent[focusOffset];
-            if (!nextCharacter) {
-                focusNode = nextLeaf(focusNode, this.editable);
+            const nextNode = nextLeaf(focusNode, this.editable);
+            if (!nextCharacter && nextNode && closestBlock(nextNode) === closestBlock(focusNode)) {
+                focusNode = nextNode;
                 focusOffset = 0;
                 nextCharacter = focusNode && focusNode.textContent[focusOffset];
```

Comment:
And so now you have the guarantee that `nextNode` is defined so `focusNode` is as well at this point, and this check can go.

Reply:
Should be good now.